packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -943,10 +943,50 @@ class RenderObjectToWidgetElement<T extends RenderObject> extends RootRenderObje
   }
 }
 
+/// Base class that provide necessary information to register a ServiceExtension.
+abstract class ExtensionRegistrarBase {
```

Comment:
Seems like this should be in the foundation library? I don't see what it has to do with widgets.